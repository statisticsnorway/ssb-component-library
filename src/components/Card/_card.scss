.ssb-card {
  @include roboto;

  .clickable {
    @include focus-marker;
    @include reset-button;

    .card-image {
      display: flex;

      img {
        height: 214px;
        width: 380px;
      }
    }

    &.left-orientation {
      display: inline-flex;
      flex-direction: row;
      width: 100%;

      .card-content {
        height: 214px;
        width: 100%;
      }

      @media #{$mobile} {
        flex-direction: column;

        .card-content {
          height: 100%;
          max-width: 380px;
        }

        .card-image {
          width: 100%;
        }

        img  {
          max-width: 380px;
          width: 100%;
        }
      }
    }

    &.top-orientation {
      display: inline-flex;
      flex-direction: column;

      .card-content {
        max-width: 380px;
        width: 100%;
      }

      @media #{$mobile} {
        width: 100%;

        .card-image {
          width: 100%;
        }

        img  {
          max-width: 380px;
          width: 100%;
        }

      }
    }

    &:hover {
      text-decoration: none;

      &.left-orientation {

        .card-content {
          border-left: 5px solid $ssb-green-4;
          padding-left: 16px;
          @media #{$mobile} {
            border-left: 1px solid $ssb-green-4;
            border-top: 5px solid $ssb-green-4;
            padding-left: 20px;
            padding-top: 15px;

            &.with-image {
              padding-top: 16px;
            }
          }
        }
      }

      &.top-orientation {

        .card-content {
          border-top: 5px solid $ssb-green-4;

          &.with-image {
            padding-top: 16px;
          }
        }
      }

      .card-content {
        border: 1px solid $ssb-green-4;

        &:not(.with-image) {
        border-top: 5px solid $ssb-green-4;
        }

        &.with-image {

          .card-title {
            background-color: $ssb-green-4;
            border-bottom: none;
            color: $ssb-white;
            margin-bottom: 10px;
          }
        }

        .arrow-icon {
          border: 2px solid $ssb-green-4;
          border-radius: 50%;
          margin-bottom: 1px;
          margin-top: 13px;
          margin-left: 1px;
          padding: 1px;
        }

        .card-action {
          align-items: center;
          background-color: $ssb-green-4;
          display: flex;
          width: fit-content;

          .arrow-icon {
            border: none;
            color: $ssb-white;
            margin-top: 0;
            padding: 0;
          }

          .href-text {
            color: $ssb-white;
          }
        }

      }
    }
  }

  .card-content {
    background: $ssb-white;
    border: 1px solid $ssb-dark-2;
    border-top: 5px solid $ssb-green-3;
    box-sizing: border-box;
    cursor: pointer;
    display: inline-block;
    flex-direction: column;
    padding: 20px;
    width: 100%;

    &.with-image {
      border-top: 1px solid $ssb-dark-2;

      .card-title {
        border-bottom: 2px solid $ssb-green-4;
        display: table;
      }
    }

    &.profiled {
      align-items: center;
      display: inline-flex;
      text-align: center;

      .card-title {
        font-size: 28px;
      }

      .card-icon {
        margin-bottom: 20px;
        max-height: 132px;
        max-width: 132px;
        width: 132px;
      }
    }

    .arrow-icon {
      box-sizing: initial;
      color: $ssb-green-4;
      display: block;
      margin-top: 12px;
      padding: 4px;
    }

    .card-action {
      @include open-sans;
      align-items: center;
      display: flex;
      margin-top: 12px;

      .arrow-icon {
        margin-top: 0;
        padding: 0;
      }

      .href-text {
        color: $ssb-dark-6;
        font-size: 16px;
        font-weight: bold;
        padding: 4px;
      }
    }

    .card-icon {
      margin-bottom: 16px;
      max-height: 28px;
      max-width: 28px;
      width: 28px;
    }

    .card-subtitle {
      @include open-sans;
      font-size: 14px;
      margin-bottom: 8px;
    }

    .card-title {
      display: block;
      font-size: 20px;
      font-weight: bold;
      margin-bottom: 8px;
    }
  }

  .download-section {
    @include open-sans;
    align-items: center;
    border: 1px solid $ssb-dark-2;
    cursor: pointer;
    display: flex;
    height: 60px;
    max-width: 380px;
    text-decoration: none;

    &:focus {
      @include focus-ring;
    }

    &:hover {
      border: 1px solid $ssb-green-3;
    }

    .download-icon {
      color: $ssb-green-4;
      margin: 0 18px;
    }

    span {
      color: $ssb-dark-6;
      font-size: 16px;
    }
  }
}
