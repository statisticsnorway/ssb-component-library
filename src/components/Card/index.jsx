import React from 'react';
import PropTypes from 'prop-types';
import { Download } from 'react-feather';

const Card = ({
	centered, children, downloadText, fileLocation, image, imagePlacement, onClick,
}) => (
	<div className="ssb-card">
		<button
			className={`clickable ${imagePlacement === 'left' ? 'left-orientation' : 'top-orientation'}`}
			onClick={onClick}
		>
			{image}
			<div className={`card-content${centered ? ' centered' : ''}${image ? ' with-image' : ''}`}>

				{ children }
			</div>
		</button>
		{fileLocation && (
			<a download href={fileLocation} className="download-section">
				<Download className="download-icon" />
				<span>{downloadText}</span>
			</a>
		)}
	</div>
);

Card.defaultProps = {
	centered: false,
	downloadText: 'Last ned',
	imagePlacement: 'top',
	onClick: () => {},
};

Card.propTypes = {
	centered: PropTypes.bool,
	children: PropTypes.node.isRequired,
	downloadText: PropTypes.string,
	fileLocation: PropTypes.string,
	image: PropTypes.element,
	imagePlacement: PropTypes.oneOf(['left', 'top']),
	onClick: PropTypes.func,
};

export default Card;
