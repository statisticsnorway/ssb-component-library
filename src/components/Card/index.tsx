import React, { type ReactNode } from 'react'
import { ArrowRight, Download, ExternalLink } from 'react-feather'

interface CardProps {
  ariaLabel?: string
  children?: ReactNode
  className?: string
  downloadText?: string
  external?: boolean
  fileLocation: string
  href: string
  hrefText?: string
  icon?: ReactNode
  image?: ReactNode
  imagePlacement?: 'left' | 'top'
  profiled?: boolean
  subTitle?: string
  title?: string
}

const Card: React.FC<CardProps> = ({
  ariaLabel,
  children,
  className,
  external,
  downloadText = 'Last ned',
  fileLocation,
  href,
  hrefText,
  icon,
  image,
  imagePlacement = 'top',
  profiled = false,
  subTitle,
  title,
}) => (
  <div className={`ssb-card${className ? ` ${className}` : ''}`}>
    <div
      className={`clickable ${imagePlacement === 'left' ? 'left-orientation' : 'top-orientation'}`}
      aria-label={ariaLabel}
    >
      {image && <div className='card-image'>{image}</div>}

      <div
        className={`card-content${image ? ' with-image' : ''}${profiled ? ' profiled' : ''}${external ? ' external' : ''}`}
      >
        {icon && <div className='card-icon'>{icon}</div>}
        {subTitle && <div className='card-subtitle'>{subTitle}</div>}
        {title && (
          <a
            href={href}
            className='card-title'
            target={external ? '_blank' : undefined}
            rel={external ? 'noreferrer' : undefined}
          >
            {title}
          </a>
        )}

        {children && <div className='card-text'>{children}</div>}

        {!image &&
          !hrefText &&
          (external ? (
            <ExternalLink className='arrow-icon' size={22} aria-hidden='true' />
          ) : (
            <ArrowRight className='arrow-icon' size={22} aria-hidden='true' />
          ))}

        {!title && !image && hrefText && (
          <a
            className='card-action'
            href={href}
            target={external ? '_blank' : undefined}
            rel={external ? 'noreferrer' : undefined}
          >
            {external ? (
              <ExternalLink className='arrow-icon' size={16} aria-hidden='true' />
            ) : (
              <ArrowRight className='arrow-icon' size={16} aria-hidden='true' />
            )}
            <div className='href-text'>{hrefText}</div>
          </a>
        )}

        {title && !image && hrefText && (
          <div className='card-action'>
            {external ? (
              <ExternalLink className='arrow-icon' size={16} aria-hidden='true' />
            ) : (
              <ArrowRight className='arrow-icon' size={16} aria-hidden='true' />
            )}
            <div className='href-text'>{hrefText}</div>
          </div>
        )}
      </div>
    </div>
    {fileLocation && (
      <a download href={fileLocation} className='download-section'>
        <Download className='download-icon' size={22} />
        <span>{downloadText}</span>
      </a>
    )}
  </div>
)

export default Card
