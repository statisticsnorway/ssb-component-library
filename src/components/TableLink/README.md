TableLink
========

> TableLink used when linkin to table
### Usage


#### HTML

```html
<div className="ssb-table-link">
    <a href="">
        <div className="tl-icon">
            <i>{feather.arrowRight 22px}</i>
        </div>
        <div className="tl-title">
            Tabell 1
            <span className="tl-text">
                Gjennomsnittspris i fritt salg på eid tomt i 100-meterbeltet. 2015-2108.
            </span>
        </div>
    </a>
</div>
```

#### React

```jsx harmony
<TableLink title="Tabell 1" href="" hrefText="Gjennomsnittspris i fritt salg på eid tomt i 100-meterbeltet. 2015-2108." />
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| className   | string | Optional container class|
| href | Required string| Destination for navigation |
| hrefText | Required string | Link text |
| hrefTitle | string | The hrefTitle attribute is used to provide additional information |
| isExternal | Optional boolean | Will make the link open in new tab. Defaults to false |
| tabIndex | number | Tab index for focus |
| title | Required string | Title for tablelink |
