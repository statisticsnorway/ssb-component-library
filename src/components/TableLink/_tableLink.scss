.ssb-table-link {
  display: flex;

  a {
    @include focus-marker;
    display: flex;
    text-decoration: none;
  }

  .tl-icon {
    color: $ssb-green-4;
    display: inline-flex;
    float: left;
    height: 22px;
    margin-right: 20px;
    margin-top: 4px;
    margin-left: 4px;
    width: 22px;
  }

  .tl-title {
    @include roboto;
    color: $ssb-green-4;
    display: inline-block;
    font-weight: bold;
    font-size: 20px;
  }

  .tl-text {
    @include roboto;
    color: $ssb-dark-6;
    display: block;
    font-weight: bold;
    font-size: 20px;
    line-height: 1.6;
    margin-top: 4px;
  }

  &:hover, &:focus {

    .tl-icon {
      border: 2px solid $ssb-green-4;
      border-radius: 50%;
      margin-top: 0;
      margin-left: 0;
      margin-right: 16px;
      padding: 2px;
    }

    .tl-text {
      color: $ssb-green-4;
    }
  }
}
