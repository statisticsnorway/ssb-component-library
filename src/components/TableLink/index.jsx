import React from 'react';
import PropTypes from 'prop-types';
import { ArrowRight } from 'react-feather';

const TableLink = ({
	className, href, hrefText, hrefTitle, isExternal, tabIndex, title,
}) => (
	<div className={`ssb-table-link${className ? ` ${className}` : ''}`}>
		<a href={href} tabIndex={tabIndex} target={isExternal ? '_blank' : undefined} title={hrefTitle}>
			<div className="tl-icon">
				<ArrowRight size={22} />
			</div>
			<div className="tl-title">
				{title}
				<span className="tl-text">
					{hrefText}
				</span>
			</div>
		</a>
	</div>
);

TableLink.defaultProps = {
	isExternal: false,
	title: 'Tabell',
};

TableLink.propTypes = {
	className: PropTypes.string,
	href: PropTypes.string.isRequired,
	hrefText: PropTypes.string.isRequired,
	hrefTitle: PropTypes.string,
	isExternal: PropTypes.bool,
	tabIndex: PropTypes.number,
	title: PropTypes.string,
};

export default TableLink;
