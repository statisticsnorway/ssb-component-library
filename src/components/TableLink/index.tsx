import React from 'react'
import { ArrowRight } from 'react-feather'

interface TableLinkProps {
  className?: string
  description: string
  href: string
  isExternal?: boolean
  tabIndex?: number
  text: string
  title?: string
}

const TableLink: React.FC<TableLinkProps> = ({
  className,
  description,
  href,
  isExternal = true,
  tabIndex,
  text,
  title,
}) => (
  <a
    className={`ssb-table-link${className ? ` ${className}` : ''}`}
    href={href}
    tabIndex={tabIndex}
    target={isExternal ? '_blank' : undefined}
    rel={isExternal ? 'noopener noreferrer' : undefined}
    title={title}
  >
    <div className='tl-icon'>
      <ArrowRight size={22} />
    </div>
    <div className='tl-info'>
      <span className='tl-text'>{text}</span>
      <span className='tl-description'>{description}</span>
    </div>
  </a>
)

export default TableLink
