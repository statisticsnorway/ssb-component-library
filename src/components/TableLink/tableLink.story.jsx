import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import TableLink from './index';

storiesOf('TableLink', module).addDecorator(centered)
	.add('default', () => (
		<div>
			<TableLink title="Tabell 1" href="" hrefText="Gjennomsnittspris i fritt salg på eid tomt i 100-meterbeltet. 2015-2108." hrefTitle="Ekstra tekst" />
			<br />
			<TableLink title="Tabell 2" href="" hrefText="Gjennomsnittspris i fritt salg på eid tomt i 100-meterbeltet. 2015-2108." />
		</div>
	));
