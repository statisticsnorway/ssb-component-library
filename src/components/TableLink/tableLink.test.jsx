import React from 'react';
import {shallow} from 'enzyme';
import {Â ArrowRight } from 'react-feather';
import TableLink from './index';
import Card from "../Card";

describe('TableLink component', () => {
    test('Matches the snapshot', () => {
        const wrapper = shallow(<TableLink title="Tabell 1" href="" hrefText="TableLink" />);
        expect(wrapper).toMatchSnapshot();
    });
    test('Toggles classNames correctly', () => {
        const wrapper = shallow(<TableLink className="mb-2" title="Tabell 1" href="" hrefText="TableLink" />);
        expect(wrapper.hasClass('mb-2')).toEqual(true);
    });
    test('Render link Text', () => {
        const wrapper = shallow(<TableLink title="Tabell 1" href="" hrefText="TableLink text" />);
        expect(wrapper.find('.tl-text').render().text()).toBe('TableLink text');
    });

});
