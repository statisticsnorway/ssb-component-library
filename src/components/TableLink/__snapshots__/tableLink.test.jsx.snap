// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableLink component Matches the snapshot 1`] = `
<div
  className="ssb-table-link"
>
  <a
    href=""
  >
    <div
      className="tl-icon"
    >
      <ArrowRight
        color="currentColor"
        size={22}
      />
    </div>
    <div
      className="tl-title"
    >
      Tabell 1
      <span
        className="tl-text"
      >
        TableLink
      </span>
    </div>
  </a>
</div>
`;
